### PR TITLE
build: migrate storybook to v7.2

### LINKED ISSUE #2212: Better take advantage of storybook
Follow up to #2163.

- [ ] Upgrade from storybook 6.4.19 to latest storybook ~(>=6.5.14)~ 7+ (_updated on 27.07.23_)
    - ℹ️ last time we tried, there were a few issues. After upgrading the libraries, try to remove the `node_modules` folder, `npm install` again, and run telepresence to connect to a deployment
- [ ] Decide on and implement the best testing strategy (see below)
- [ ] Move the style-guide to storybook

We should also figure out how we want to take advantage of storybook for testing. Here are two options.

1. Jest-based: using storybook/testing-library integration https://storybook.js.org/blog/interaction-testing-with-storybook/ 
2. Cypress-based: start storybook (similar to how we start the UI) and click around the components

### PR DESCRIPTION
**Pull request by andre-code**
Closes #2212 

/deploy #persist renku=renku-ui/3.11-acceptance-tests

### CODE REVIEW

**Comment by leafty** [comment]
Thank you @andre-code :raised_hands: 

Some stories seem to crash:
image

"Visibility filter" > Click on any option

Should this be fixed here or in a later PR?

**Review by lorenzo-cavazzi — COMMENTED** [review]
I actually added my comments on #2744 

Is it ok with you if we close this PR and continue there directly?

**Comment by andre-code** [comment]
> Thank you @andre-code 🙌
> 
> Some stories seem to crash: image
> 
> "Visibility filter" > Click on any option
> 
> Should this be fixed here or in a later PR?

No, the broken stories were fixed here  https://github.com/SwissDataScienceCenter/renku-ui/pull/2744

File: client/src/notebooks/Notebooks.container.js [hunk 1]
```diff
@@ -742,6 +744,8 @@ class StartNotebookServer extends Component {
     this.state.rtkQuerySubscriptions.push(getResourcePools.unsubscribe);
 
     const { data: resourcePools } = await getResourcePools;
+    // TODO: Do not mutate state directly. Use setState()
+    // eslint-disable-next-line react/no-direct-mutation-state
```
[the remark this answers is not in the document]
Comment (andre-code) [reply client/src/notebooks/Notebooks.container.js:748]: yes, the storybook eslint complain about it and do not start the storybook, if the file will be dropped better to leave the comment

**Comment by andre-code** [comment]
> I actually added my comments on #2744
> 
> Is it ok with you if we close this PR and continue there directly?

thanks for your review @lorenzo-cavazzi and @leafty 
the intention of this PR is to get storybook working again using a more updated version, we can use this PR as a feature branch, but please evaluate the PR with the intention of the PR, no broken component review or documentation is included at this time

**Comment by leafty** [comment]
> > I actually added my comments on #2744
> > Is it ok with you if we close this PR and continue there directly?
> 
> thanks for your review @lorenzo-cavazzi and @leafty the intention of this PR is to get storybook working again using a more updated version, we can use this PR as a feature branch, but please evaluate the PR with the intention of the PR, no broken component review or documentation is included at this time

Fair point, but it was not clear from the PR description.

File: client/.storybook/store.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { createStore } from "../src/utils/helpers/EnhancedState.js";
+import { createStore } from "../src/utils/helpers/EnhancedState.ts";
```
Comment (leafty) [inline client/.storybook/store.js:1]:
```suggestion
import { createStore } from "../src/utils/helpers/EnhancedState";
```

**Review by leafty — APPROVED** [review]

**Review by leafty — APPROVED** [review]

**Pull request merged**